File: algorithms/merge-sorted-array/rust/merge-sorted-array.rs
impl Solution {
    pub fn merge(nums1: &mut Vec<i32>, m: i32, nums2: &mut Vec<i32>, n: i32) {
        let mut a = m as usize;
        let mut b = n as usize;
        let mut c = (m + n) as usize;

        while a > 0 && b > 0 {
            if nums1[a - 1] < nums2[b - 1] {
                nums1[c - 1] = nums2[b - 1];
                c -= 1;
                b -= 1;
            } else {
                nums1[c - 1] = nums1[a - 1];
                c -= 1;
                a -= 1;
            }
        }
        while b > 0 {
            nums1[c - 1] = nums2[b - 1];
            c -= 1;
            b -= 1;
        }
    }
}

File: algorithms/merge-sorted-array/merge-sorted-array.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function (nums1, m, nums2, n) {
  let a = m - 1;
  let b = n - 1;
  let c = m + n - 1;

  while (a >= 0 && b >= 0) {
    if (nums1[a] < nums2[b]) {
      nums1[c--] = nums2[b--];
    } else {
      nums1[c--] = nums1[a--];
    }
  }
  while (b >= 0) {
    nums1[c--] = nums2[b--];
  }
};
